src/objectExplorer/objectExplorerProvider.ts

```diff
@@ -49,11 +49,11 @@ export class ObjectExplorerProvider implements vscode.TreeDataProvider<any> {
         return undefined;
     }
 
-    public removeObjectExplorerNode(node: TreeNodeInfo, isDisconnect: boolean = false): Promise<void> {
+    public async removeObjectExplorerNode(node: TreeNodeInfo, isDisconnect: boolean = false): Promise<void> {
```

Comment:
>async [](start = 11, length = 5)

Why are you making these async? It's not necessary since removeObjectExplorerNode already returns a promise

Reply:
its cleaner to use async/await

Reply:
But you're not awaiting anything - so this is adding nothing. Returning a Promise happens regardless and making it async or not doesn't change that. Anyways it's not a huge deal since it just doesn't do anything - but you should really consider cleaning up pointless stuff like this unless it's needed since it just adds extra overhead.

Reply:
https://github.com/microsoft/vscode-mssql/blob/master/src/controllers/mainController.ts#L176

I need to await here - https://github.com/microsoft/vscode-mssql/blob/master/src/controllers/mainController.ts#L176

I don't want to have objectExplorerService to be exposed directly to the controller, so I call it via the provider. Please look at the usage of code before calling it useless.

Reply:
You can await a function that returns a promise. It being asyc doesn't affect that. If your method signature was

```
public removeObjectExplorerNode(node: TreeNodeInfo, isDisconnect: boolean = false): Promise<void> {
```

it would be exactly the same logic-wise as it is now.